src/main/java/org/openrewrite/java/testing/assertj/testng/TestNgAssertEqualsToAssertThat.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2021 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.testing.assertj.testng;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.TypeUtils;
+
+import java.util.List;
+
+public class TestNgAssertEqualsToAssertThat
```

Comment:
I could be missing something! But couldn't this be a [Refaster Teamplate Recipe](https://docs.openrewrite.org/authoring-recipes/types-of-recipes#refaster-template-recipes)?
I think this could work quite generically:
```java
public class MigrateToAssertEquals {
    @BeforeTemplate
    void before(Object actual, Object expected) {
        Assert.assertEquals(actual, expected);
    }
    @AfterTemplate
    void after(Object actual, Object expected) {
        Assertions.assertThat(actual).isEqualsTo(expected);
    }
}
```

Reply:
> couldn't this be a [Refaster Teamplate Recipe](https://docs.openrewrite.org/authoring-recipes/types-of-recipes#refaster-template-recipes)?

@MBoegers guess that would work fine. On the other hand @ssheikin's explicit implementation is a bit more performant, as it returns the method directly, while the code generated by refaster will continue visiting any tree node below it. One could ask though why refaster does that in the first place :wink:

Reply:
Indeed, from a performance point recipe can benefit from an imperative style. But from a maintainability and understandability point refaster template recipes win, at least for me.
Let's shift the discussion about _Refaster Template Recipe speed-ups_ to the Slack ;)

Reply:
@MBoegers have created https://github.com/openrewrite/rewrite-templating/issues/95 as a followup.